### ISSUE TITLE
Internal Seafile Links - Feature Proposal \ Question

### ISSUE THREAD

**Issue by GITchristoph** [haiwen/seafile#1059]
Dear all,
Is it possible to create "internal seafile links"?
And if yes, would you agree that it's a useful feature and consider to implement it?

Let me try to explain with an example:

> User A makes an update of a drawing in a shared library.
> The drawing gets synced between all persons who share the library.
>
> *So far all as normal procedure, but now:*
>
> User A now creates an "internal seafile link" to that file and sends it to the relevant people.
> (Can use Email, messenger, skype, trello, slack, whatever she/he wishes and our clients are using)
> By clicking on the link, it directly opens the file in everybody's seafile folder on the computer. The file > > that has been already synced through seafile.

**So the link does not download the file through a browser but opens the file in each's personal file structure on each's personal computer directly.**

We would assume that it would require the seafile client to be installed on that computer and that the user has permission to share the library.

**Why I ask for it?**
We could use any messaging or organiser service for distributing internal seafile links and informing about file changes. We can be flexible to what our clients use. The links can also be incorporated into any project management software. It would help a lot to structure revisions and status of projects and e.g. drawing development without creating the data double or triple.

It would be quite save to share file locations via services running on other external servers. Because the actual data stays on our own server in the Seafile with permissions under our control.

We experience with colleagues and clients, that after a drawing or excel file or another document has been revised, updated or created, that then an email or some sort of message gets distributed to colleagues and clients, explaining the location of the file.
Sometimes even the file itself gets sent in an email or gets uploaded to organisation software, although it synced via the seafile client already. Sending an internal link would make that much more convenient.

I am very curious if this proposal is interesting also for other users or if there is maybe an easier way to achieve this?

We think it's a killer feature ;-)

Thanks a lot for Seafile, we love the software in our office.

KR, Chris

**Comment by freeplant** [comment]
Thanks for the suggestion. It's possible to add a special seafile link to files like seafile://path/to/file . When a user click it, it will be opened locally.

**Comment by GITchristoph** [comment]
Thanks freeplant! 
This sounds very promising but I don't get there yet.

If I want to link to a file *sample.txt* in a Library called *Shared* then the link would be:

> seafile://Shared/sample.txt  correct?

I tried it on OSX Yosemite to email myself that link but when I click on it, nothing happens. 

What am I missing?

**Comment by shoeper** [comment]
it's not implemented so far.

**Comment by GITchristoph** [comment]
Ah, sorry, I read that to quick.
Then: Yes exactly, that would be great feature, we think!

;-)

**Comment by freeplant** [comment]
;-)

**Comment by GITchristoph** [comment]
Little addon:

It would be great if the link would also work in the seafile clients on iOS and Android. 

Imagine how cool that would be! ;-)

**Comment by GITchristoph** [comment]
Hi Freeplant

I actually just saw that the Mac Version of the client supports this feature since 4.2.5.
(https://seacloud.cc/group/3/wiki/client-changelog/)

**Fantastic and Thanks a lot!**

I have a quick question:

Do you know if it would be generally feasible to make this work also on iOS / Android Devices that have a seafile client installed? 
Not "pushing", just asking if you know if the OS's would allow it.

Cheers,

Chris

**Comment by shoeper** [comment]
At least on android it is possible. But I'm quite sure that it should also 
be possible on iOS.


Am 13. Juli 2015 16:17:16 schrieb GITchristoph <notifications@github.com>:

**Comment by GITchristoph** [comment]
Thanks shoeper

Looking forward to it ;-)

**Comment by GITchristoph** [comment]
Dear all,

Is it possible that the internal link does not yet work on windows computers?

If I send a link, that's generated on a mac, to a windows computer for testing, then all that happens is, that the seafile window pops up. But the linked file never gets opened.

If the same windows computer sends a link to the mac, then it works and the linked file opens.
If the same windows computers tries to open its own created link, then it also doesn't work and all that happens is that the seafile program opens up.

Both clients are version 4.2.8. Server: 4.2.3.

Is this the current status (that windows client doesn't open these internal links yet) or is this a bug?

Greetings, Chris

**Comment by GITchristoph** [comment]
Sorry to bother, I would just really like to ask again if this feature is supposed to work on the windows client or if it isn't or if it needs an additional installation/preference on the windows side?

As you can see (from us having the idea ;-) ) we are really waiting on this feature and it would be great if we could get it to work on windows as well.

Looking forward to hear from you,

Greetings, Chris

**Comment by freeplant** [comment]
This maybe a bug. But we don't have a time to verify it and fix it yet.

**Comment by GITchristoph** [comment]
Thanks Daniel, understood.

**Comment by freeplant** [comment]
I can't reproduce the problem with my Mac and Windows. But I'm using v4.3.0 Windows client. You can try later when it is published.

**Comment by GITchristoph** [comment]
Thanks for Client 4.3.0!

Unfortunately the internal links still don't work on windows for us:

If I send an internal link from a Mac to a PC and I click on it on the PC, then all that happens is that the seafile app window opens and nothing else.

Same for links to files and folders.

**Comment by GITchristoph** [comment]
This issue is solved for us.

We used seafile as the application to open the internal seafile links in windows. 
That didn't work, it's necessary to use a browser (like iExplorer or Chrome).

Please check Forum Thread:
https://forum.seafile-server.org/t/seafile-internal-link/2588

I will try to make an entry into the manual in the near future for other users.

In any way, it seems a bit "complicated" that after clicking on the link, first a file browser opens and then the actual linked file, no?

I will play around with it further and if I should have any more ideas, I will let you know in another issue.

This one is closed because the general functionality is there.

Kind Regards, Chris
